### ISSUE TITLE
No Token!

### ISSUE THREAD

**Issue by biglynx55** [YinHangCode/homebridge-mi-airpurifier#26]
My "miio discover sync" answers as follows:

 INFO  Discovering devices. Press Ctrl+C to stop.

Device ID: 81001836
Model info: Unknown
Address: 192.168.1.85
Token: ???
Support: Unknown

I did wifi connection reset procedure - again the same. Device ID i correct because router recognized is as: zhimi-airpurifier-mc1_miio81001836

Any ideas?

Regards, Tomek

**Comment by Hinchukardt** [comment]
Hi Tomek,

Try restarting everything, your air purifier, your wifi router and the device you're running miio on.
You can also try using the mi home app during the discovery process to control the air purifier. That worked for me.

If that doesn't resolve your problem, you should go to the specific page for miio: [https://github.com/aholstenson/miio/issues](url)
This page is for issues with the homebridge-mi-airpurifier, your issue is with the miio process.

**Comment by pax0707** [comment]
Well, to get Mi token I had to install android under an emulator (and root it to get access to data folder). Google that I need an old version of Mi app (because new ones don't save tokens locally), find a way to get the db file out of the emulator and then rummage through it to get the token.

As someone said: why waste precious minutes of your life when you can automate it within a few hours.

**Comment by barabba9174** [comment]
It's the last firmware, the token isn't available from miso, it isn't specifically a homebridge-mi-airpurifier issue.

**Comment by biglynx55** [comment]
> It's the last firmware, the token isn't available from miso, it isn't specifically a homebridge-mi-airpurifier issue.

I agree. But to use homebridge I need token anyway. So question how to get it from new firmware remains still open

**Comment by MickL** [comment]
Please close this issue as it is not related to `homebridge-mi-airpurifier`. 

Anyway you can get the toking using one of the following methods: https://github.com/jghaanstra/com.xiaomi-miio/blob/master/docs/obtain_token.md

I just did it with iOS method and the latest Firmware update an Mi App.
